### ISSUE TITLE
iOS silent mode is not honored by default

### ISSUE THREAD

**Issue by kyle-sylvestre** [libsdl-org/SDL#11140]
Related #3831

Sound continues to play when silent mode is turned on. This is due to the default audio session category being AVAudioSessionCategoryPlayback. I propose we restore the default back to AVAudioSessionCategoryAmbient to avoid this problem.

https://github.com/libsdl-org/SDL/blob/9023a199dda84a2db372d415a593a39053d7db49/src/audio/coreaudio/SDL_coreaudio.m#L411

**Comment by slouken** [comment]
No, this was an intentional change. The idea is that if you're playing SDL audio, that it's important for gameplay and you actually want it to be heard. If you are playing it as ambient audio, you can set the SDL_HINT_AUDIO_CATEGORY appropriately.
